src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs

```diff
@@ -324,35 +326,45 @@ public static void AddTopLevelSuggestionsForCursorType(IntellisenseData.Intellis
             Contracts.AssertValue(callNode);
             Contracts.Assert(argPosition >= 0);
 
-            var info = intellisenseData.Binding.GetInfo(callNode);
+            CallInfo info = intellisenseData.Binding.GetInfo(callNode);
             Contracts.AssertValue(info);
-            var type = info.CursorType;
+            DType type = info.CursorType;
 
             // Suggestions are added for the error nodes in the next step.
-            if (info.Function != null &&
-                argPosition <= info.Function.MaxArity &&
-                info.Function.IsLambdaParam(argPosition) &&
-                !info.Function.HasSuggestionsForParam(argPosition) &&
-                type.IsValid)
+            if (info.Function != null && argPosition <= info.Function.MaxArity)
             {
-                if (info.Function.IsLambdaParam(argPosition) && type.ContainsDataEntityType(DPath.Root))
+                if (info.Function.IsLambdaParam(argPosition) && !info.Function.HasSuggestionsForParam(argPosition) && type.IsValid)
                 {
-                    var error = false;
-                    type = type.DropAllOfTableRelationships(ref error, DPath.Root);
-                    if (error)
+                    if (info.Function.IsLambdaParam(argPosition) && type.ContainsDataEntityType(DPath.Root))
                     {
-                        return;
+                        var error = false;
+                        type = type.DropAllOfTableRelationships(ref error, DPath.Root);
+                        if (error)
+                        {
+                            return;
+                        }
                     }
-                }
 
-                if (info.ScopeIdentifier != default)
-                {
-                    AddSuggestion(intellisenseData, info.ScopeIdentifier, SuggestionKind.Global, SuggestionIconKind.Other, type, requiresSuggestionEscaping: false);
+                    if (info.ScopeIdentifier != default)
+                    {
+                        AddSuggestion(intellisenseData, info.ScopeIdentifier, SuggestionKind.Global, SuggestionIconKind.Other, type, requiresSuggestionEscaping: false);
+                    }
+
+                    if (!info.RequiresScopeIdentifier)
+                    {
+                        AddTopLevelSuggestions(intellisenseData, type);
+                    }
                 }
 
-                if (!info.RequiresScopeIdentifier)
+                // If connector function has some suggestions, let's add them here
+                List<string> suggestions = info.Function.GetConnectorSuggestionsAsync(callNode, argPosition, CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
>CancellationToken.None [](http://example.com/codeflow?start=109&length=22)

note - if intellisense ever gets a cancellation token, we would pass it in here.

Reply:
yes intellisense is synchronous for now
there is no SuggestAsync API...
probably needs to be added